### PR TITLE
an error handling issue

### PR DESCRIPTION
**Pull request by gymad**
## Description

PR contains some fix on email client, also it contains a possible fix for  Issue-1 in task comment on scrm-20
- fix php errors and notices on email sending function
- change the error handling and response in ajax request for email sending
- handle separately the user messages (for e.g for the wron CRM settings) when the email sending

## Motivation and Context

some wrong error handling

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/Emails/include/ComposeView/EmailsComposeView.js [hunk 1]
```diff
@@ -518,6 +518,105 @@
         formData.append($(v).attr('name'), $(v).val());
       });
 
+      var ajaxEmailSendingResponse = function(data) {
+        // todo: (ref: SCRM-20) here we have to outsourcing the error handling to doing it in similar common way anywhere
```
Comment (daniel-samson) [inline modules/Emails/include/ComposeView/EmailsComposeView.js]: TODO? Should this be todo: task: ref - title?

File: modules/Emails/views/view.sendemail.php [hunk 2]
```diff
@@ -57,22 +57,30 @@ public function __construct()
...
+            );
+        } else {
+            $response = array (
+                'status' => !empty($this->bean->status) ? $this->bean->status : 'sent',
+                'messages' => $mod_strings['LBL_MESSAGE_SENT'],
+            );
+        }
+        echo json_encode($response);
+        if($err = json_last_error()) {
+            throw new Exception("trying to send an incorrect JSON response ($err)");
```
Comment (daniel-samson) [inline modules/Emails/views/view.sendemail.php]: Should this start with a capital letter?

File: modules/Emails/include/ComposeView/EmailsComposeView.js [hunk 3]
```diff
@@ -527,22 +626,16 @@
...
+        ajaxEmailSendingResponse(data);
+      }).fail(function (data) {
+        if(data) {
+          ajaxEmailSendingResponse(data);
         } else {
-          $(self).trigger("sentEmail", [self, data]);
+          // here always handle only the http problems,
+          // not anything else (for e.g email sending issues or CRM settings problems..)
+          console.error(data);
+          throw "an ajax XMLHttpRequest request failed";
```
Comment (daniel-samson) [inline modules/Emails/include/ComposeView/EmailsComposeView.js]: Should this start with "An AJAX" ?

**Comment by Mac-Rae** [comment]
Please resolve the conflicts on this pull request bringing it up to date.

**Pull request closed without merging**